source/common/http/header_utility.cc

```diff
@@ -139,7 +139,16 @@ bool HeaderUtility::matchHeaders(const HeaderMap& request_headers, const HeaderD
 
   if (!header_value.result().has_value()) {
     if (header_data.invert_match_) {
-      return header_data.header_match_type_ == HeaderMatchType::Present && header_data.present_;
+      if (Runtime::runtimeFeatureEnabled(
+              "envoy.reloadable_features.return_match_when_invert_header_match_"
+              "and_header_is_not_present")) {
```

Comment:
maybe name it as `apply_invert_match_when_header_not_present_and_non_present_match_type`. But I'm not good at naming, just a suggestion, and we can see others' opinions here.